### PR TITLE
Add --skip-failed-analyses option

### PR DESCRIPTION
**Pull request by odashi**
This PR adds ~~`skip_failures`~~ `skip_failed_analyses` flag to `Processor` and ~~`--skip-failures`~~ `--skip-failed-analyses` CLI option to provide a user-side control to choose whether all analyses should succeed or not. Each `Analysis` now returns just `AnalysisResult` or raise errors, and error handling is delegated to `Processor`.

This PR also contains several typing fixes encountered during applying this change.

### CODE REVIEW

**Review by neubig — APPROVED** [review]
The code itself looks good, but perhaps `--skip-failed-analyses` would be more clear as naming, especially as an argument to `explainaboard_main`. I'll leave it up to you whether you think changing to this would be helpful (happy to approve even if it's not changed).

**Comment by odashi** [comment]
> `--skip-failed-analyses`

SGTM. I will change this pr.

**Review by pfliu-nlp — APPROVED** [review]
Thanks, this strategy looks good to me.

**Pull request merged**
